cilium-cli/connectivity/tests/errors.go

```diff
@@ -115,11 +116,12 @@ func (n *noErrorsInLogs) Run(ctx context.Context, t *check.Test) {
 		for _, container := range info.containers {
 			id := fmt.Sprintf("%s/%s/%s (%s)", pod.Cluster, pod.Namespace, pod.Name, container)
 			t.NewGenericAction(n, id).Run(func(a *check.Action) {
-				logs, err := client.GetLogs(ctx, pod.Namespace, pod.Name, container, opts)
+				var logs bytes.Buffer
+				err := client.GetLogs(ctx, pod.Namespace, pod.Name, container, opts, &logs)
 				if err != nil {
 					a.Fatalf("Error reading Cilium logs: %s", err)
 				}
-				n.checkErrorsInLogs(id, logs, a)
+				n.checkErrorsInLogs(id, logs.String(), a)
```

Comment:
Drive-by comment: rather than converting the `bytes.Buffer` to a string and potentially inducing additional copies of the string which increase memory usage, could we instead change `checkErrorsInLogs` to work on `[]byte` which wouldn't induce these copies?

Reply:
Though this is probably a bit out of scope for this PR which deals with sysdump memory usage, so feel free to disregard.

Reply:
Hmm, fair point, I had kept the same signature for simplicity given that the `[]byte` to `string` conversion was already preset before. Without extending too much the scope of this PR, I've now changed the signature to propagate the byte slice, and perform the conversion to string directly on each splitted chunk of logs. WDYT?